File: Hooks/Hooks.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.IE;
using TechTalk.SpecFlow;

namespace SwagLabsAutomation.Hooks
{
    [Binding]
    public class Hooks
    {
        private WebDriver driver;

        [BeforeScenario]
        public void InitializeTest()
        {
            // Initialize WebDriver based on browser type (can be parameterized)
            string browserType = "chrome"; // default browser
            switch (browserType.ToLower())
            {
                case "chrome":
                    driver = new ChromeDriver();
                    break;
                case "firefox":
                    driver = new FirefoxDriver();
                    break;
                case "ie":
                    driver = new InternetExplorerDriver();
                    break;
                default:
                    driver = new ChromeDriver();
                    break;
            }

            ScenarioContext.Current["WebDriver"] = driver;
        }

        [AfterScenario]
        public void TearDownTest()
        {
            if (driver != null)
            {
                driver.Quit();
            }
        }
    }
}

File: src/test/java/hooks/Hooks.java
import io.cucumber.java.After;
import io.cucumber.java.Before;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.ie.InternetExplorerDriver;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.safari.SafariDriver;

public class Hooks {
    private WebDriver driver;

    @Before
    public void initializeTest() {
        String browser = System.getProperty("browser", "chrome");
        switch (browser.toLowerCase()) {
            case "chrome":
                System.setProperty("webdriver.chrome.driver", "path_to_chromedriver");
                driver = new ChromeDriver();
                break;
            case "firefox":
                System.setProperty("webdriver.gecko.driver", "path_to_geckodriver");
                driver = new FirefoxDriver();
                break;
            case "ie":
                System.setProperty("webdriver.ie.driver", "path_to_IEDriverServer");
                driver = new InternetExplorerDriver();
                break;
            case "edge":
                System.setProperty("webdriver.edge.driver", "path_to_EdgeDriver");
                driver = new EdgeDriver();
                break;
            case "safari":
                driver = new SafariDriver();
                break;
            default:
                throw new IllegalArgumentException("Invalid browser specified: " + browser);
        }
    }

    @After
    public void tearDownTest() {
        if (driver != null) {
            driver.quit();
        }
    }
}
